NExtends/Collections/DictionaryListMapper.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NExtends.Collections
+{
+	/// <summary>
+	/// Helper class mapping entries of an IList&lt;<typeparamref name="V"/>&gt; to an IDictionary with configurable key values
+	/// </summary>
+	/// <example>
+	/// <code>
+	///		var test = new string[] { "entry1", "entry2" };
+	///
+	///		var entries = new DictionaryListMapEntry[] {
+	///			new  DictionaryListMapEntry(0, "lala"),
+	///			new  DictionaryListMapEntry(1, "lulu")
+	///		};
+	///		var m = new DictionaryListMapper&lt;string&gt;(test, entries);
+	///
+	///		Assert.IsTrue(m["lala"] == "entry1");
+	///		Assert.IsTrue(m["lulu"] == "entry2");
+	/// </code>
+	/// </example>
+	/// <typeparam name="V"></typeparam>
+	public class DictionaryListMapper<V> : IDictionary<string, V>
+	{
+		IList<V> CollectionToMap;
+		IEnumerable<IDictionaryListMapEntry> MapEntries;
+		ICollection<string> MapEntriesKeys;
+		ICollection<int> MapEntriesIndexes;
+
+		Dictionary<string, IDictionaryListMapEntry> Map;
+
+		public DictionaryListMapper(IList<V> collectionToMap, IEnumerable<IDictionaryListMapEntry> mapEntries)
+			: this(collectionToMap)
+		{
+			SetMapEntries(mapEntries);
+		}
+
+		protected DictionaryListMapper(IList<V> collectionToMap)
+		{
+			this.CollectionToMap = collectionToMap;
+		}
+
+		protected void SetMapEntries(IEnumerable<IDictionaryListMapEntry> mapEntries)
+		{
+			this.MapEntries = mapEntries;
+			this.MapEntriesKeys = this.MapEntries.Select(me => me.Key).ToList();
+			this.MapEntriesIndexes = this.MapEntries.Select(me => me.Index).ToList();
+
+			Map = new Dictionary<string, IDictionaryListMapEntry>();
+			foreach (var entry in mapEntries)
+				Map.Add(entry.Key, entry);
+		}
+
+		public void Add(string key, V value)
+		{
+			throw new NotImplementedException();
+		}
+
+		public bool ContainsKey(string key)
+		{
+			return MapEntriesKeys.Contains(key);
+		}
+
+		public ICollection<string> Keys
+		{
+			get { return MapEntriesKeys; }
+		}
+
+		public bool Remove(string key)
+		{
+			throw new NotImplementedException();
+		}
+
+		public bool TryGetValue(string key, out V value)
+		{
+			bool succes = false;
+
+			value = default(V);
+			try
+			{
+				int idx = Map[key].Index;
+				value = CollectionToMap[idx];
+
+				succes = true;
+			}
+			catch
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Handle the exception or explain in a comment why it can be ignored. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=csharpsquid%3AS2486)